File: src/main/java/org/example/expert/domain/todo/entity/Log.java
package org.example.expert.domain.todo.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "log")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Log {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String message;

    private String requestIp;

    private String requestUser;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    public void prePersist() {
        this.createdAt = LocalDateTime.now();
    }

    public Log(String message, String requestIp, String requestUser) {
        this.message = message;
        this.requestIp = requestIp;
        this.requestUser = requestUser;
    }
}

File: src/main/java/org/example/expert/domain/todo/entity/Log.kt
package org.example.expert.domain.todo.entity

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name="log")
class Log (
    var message: String,
    var requestIp: String,
    var requestUser: String,

    @Column(nullable = false, updatable = false)
    var createdAt: LocalDateTime? = null,
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @PrePersist
    fun prePersist() {
        this.createdAt = LocalDateTime.now()
    }
    protected constructor() :this("","","")
}